content-review/impl/urkund/src/main/java/org/sakaiproject/contentreview/urkund/UrkundReviewServiceImpl.java

```diff
@@ -620,10 +633,22 @@ public String getIconCssClassforScore(int score, String contentId) {
 		}
 	}
 
+	public ResourceLoader getResourceLoader() {
```

Comment:
Almost - In BaseContentReviewSerivce, I'd do:
return new ResourceLoader(getServiceName().toLowerCase());
but TurnitinOC's service name is 'Turnitin' which is the same as Turnitin legacy. Their properties files shared the same name before, which caused funny bugs.
So, I'd have to do something like:
return new ResourceLoader(getResourceBundleName());
and then implement getResourceBundleName() in each service to return the unique resource file's name, which is essentially the same thing, plus a few lines in BaseContentReviewService. I kinda like the current code better

Reply:
Wouldn't you only need to override the `getResourceBundleName()` method in the child-classes to return just the string? Then only the super-class would define the several `getResourceLoader()` methods, calling `getResourceBundleName()` as necessary. I think it would end up with less code duplication between the child-classes (only one method to implement, returning a string, rather than several `getResourceLoader()` methods).

Reply:
Oh yeah I think I see what you mean, like this:
BaseContentReviewService:
public ResourceLoader getResourceLoader() {
    return new ResourceLoader(getResourceBundleName())
}
public getResourceBundleName(){
    return getServiceName().toLowerCase();
}

ContentReviewServiceTurnitinOC:
public getResourceBundleName() { return "turnitin-oc"; }

Reply:
Yep, you got it. You may also want to define `getResourceBundleName()` as `abstract` in the super-class to force any future child-classes to implement it explicitly.

Reply:
We can get away without abstract because we're shifting the implementation to getServiceName(); only TurnitinOC has to override it because its bundle file doesn't match its getServiceName()